### PR TITLE
[MAINTENANCE] Notify on build-n-publish failure.

### PR DESCRIPTION
**Pull request by billdirks**
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -357,33 +357,6 @@ jobs:
         # Due to the relatively small number of usage_stats tests, we deem it appropriate to test them in their entirely through pytest
         run: pytest --no-sqlalchemy --aws-integration -v tests/integration/usage_statistics
 
-  notify_on_failure:
```
Comment (billdirks) [inline .github/workflows/ci.yml:360]: I moved this to the bottom.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -427,3 +400,33 @@ jobs:
...
+      docs-snippets,
+      unit-tests,
+      cloud-tests,
+      marker-tests,
+      py38-min-versions,
+      py39-min-versions,
+      py310-min-versions,
+      import_gx,
+      usage_stats_integration,
+      build-n-publish
```
Comment (billdirks) [inline .github/workflows/ci.yml:417]: This block is identical to above except for adding `build-n-publish`.

**Review by Kilo59 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -427,3 +400,33 @@ jobs:
...
+    if: always() && !cancelled() && contains(needs.*.result, 'failure') && github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Scheduled CI job failure
+        id: slack
+        uses: slackapi/slack-github-action@v1.24.0
+        with:
+          payload: |
+            {
+              "event_name": "${{ github.event_name }}",
```
Comment (billdirks) [inline .github/workflows/ci.yml:428]: I also added the `event_name` to the message.

**Review by Shinnnyshinshin — APPROVED** [review]
✅

**Pull request merged**